### ISSUE TITLE
`cargo vendor` blows up.

### ISSUE THREAD

**Issue by sjmackenzie** [input-output-hk/jormungandr#323]
**Describe the bug**
When integrating `jormungandr` into vendor projects, duplicate dependencies collide.

**Steps To Reproduce**
1. run `cargo vendor` in the base directory

**Actual behaivour**
```
$ cargo vendor 
error: failed to sync

Caused by:
  found duplicate version of package `tower-util v0.1.0` vendored from two sources:

        source 1: https://github.com/tower-rs/tower#4d6d2c85
        source 2: registry https://github.com/rust-lang/crates.io-index
```

**Expected behavior**
The command runs without error

**Comment by sjmackenzie** [comment]
`git bisect` says this:
```
Updated 1 path from the index
311abe4f69bdefb300459f904335b1871bc6cc65 is the first bad commit
commit 311abe4f69bdefb300459f904335b1871bc6cc65
Author: Eelco Dolstra <edolstra@gmail.com>
Date:   Mon Apr 29 11:22:33 2019 +0200

    Fix build against rust-cardano master

:100644 100644 345a309ca10601e62145811297501da2881fa0a1 13f06c88954cee97380c4793f9f2a060eabe87a0 M   Cargo.lock
:160000 160000 45237c435c17d50773bf53a4c5f70761f8cf5643 f1a008bea3638b2c34519f4b1d378bfbf0206b9b M   cardano-deps
:040000 040000 54e251bf667a078ae16c8f5af9d4510c2a93df20 306283144c39cac491ed1a1c846cb158634af47e M   src
```
cc @edolstra

**Comment by sjmackenzie** [comment]
Closing, as it's not really part of the current development workflow. Though it breaks `buildRustPackage` in nixpkgs.

**Comment by sjmackenzie** [comment]
The problem originates from this commit https://github.com/tower-rs/tower-grpc/commit/92b34fd421fda40c53d4e3f7e160287ced06dd50 @mzabaluev if you could update https://github.com/input-output-hk/rust-cardano/blob/master/network-grpc/Cargo.toml#L29 to at least this commit that's be superb!

**Comment by mzabaluev** [comment]
I'm trying to understand whether `cargo vendor` is more fragile than the regular build with unmodified `Cargo.lock`, or is it just another time to update tower-grpc deps again.

**Comment by sjmackenzie** [comment]
oh I'm sure things will be a lot better once tower stabilizes. For now I've got a patchwork system dealing with the jormungandr + rust-cardano ( now + tower ) churn. `cargo vender` is very specific to my nix setup. So it's of no real concern to you guys. Though this update would be very welcome indeed. I don't want to make a patch cause I know this is your domain and don't want to tread on toes (make conflicts).

**Comment by mzabaluev** [comment]
Will be finally fixed by #326.
